src/main/java/uk/gov/hmcts/reform/prl/handlers/TransferToAnotherCourtEventHandler.java

```diff
@@ -0,0 +1,158 @@
+                        emailTemplateNames,
+                        transferCaseContentProvider.buildCourtTransferEmailCitizen(caseData, value,false),
+                        LanguagePreference.getPreferenceLanguage(caseData)
+                    ));
+        }
+    }
+
+    private void sendEmailToRespondentSolicitor(CaseData caseData, EmailTemplateNames emailTemplateName) {
+        List<Map<String, List<String>>> respondentSolicitors = caseData
+            .getRespondents()
+            .stream()
+            .map(Element::getValue)
+            .filter(i -> YesNoDontKnow.yes.equals(i.getDoTheyHaveLegalRepresentation()))
+            .map(i -> {
+                Map<String, List<String>> temp = new HashMap<>();
+                temp.put(i.getSolicitorEmail(), List.of(
+                    i.getRepresentativeFirstName() + " " + i.getRepresentativeLastName(),
+                    i.getFirstName() + " " + i.getLastName()
+                ));
+                return temp;
+            })
+            .collect(Collectors.toList());
+
+        for (Map<String, List<String>> resSols : respondentSolicitors) {
+            String solicitorEmail = resSols.keySet().toArray()[0].toString();
+            emailService.send(
+                solicitorEmail,
+                emailTemplateName,
+                transferCaseContentProvider.buildCourtTransferEmailSolicitor(caseData, resSols.get(solicitorEmail).get(0)),
+                LanguagePreference.english
+            );
+        }
+    }
+
+    private void sendEmailToApplicant(CaseData caseData, EmailTemplateNames emailTemplateName) {
+        Map<String, String> applicantEmails = caseData.getApplicants().stream()
+            .map(Element::getValue)
+            .filter(applicant -> !CaseUtils.hasLegalRepresentation(applicant)
+                && Yes.equals(applicant.getCanYouProvideEmailAddress()))
+            .collect(Collectors.toMap(
+                PartyDetails::getEmail,
+                party -> party.getFirstName() + " " + party.getLastName(),
+                (x, y) -> x
+            ));
+
+        if (!applicantEmails.isEmpty()) {
+            applicantEmails.forEach(
+                (key, value) ->
+                    emailService.send(
+                        key,
+                        emailTemplateName,
+                        transferCaseContentProvider.buildCourtTransferEmailCitizen(caseData, value,false),
+                        LanguagePreference.getPreferenceLanguage(caseData)
+                    ));
+        }
+    }
+
+    private void sendEmailToApplicantSolicitor(CaseData caseData, EmailTemplateNames emailTemplateName) {
+        Map<String, String> applicantSolicitors = caseData
+            .getApplicants()
+            .stream()
+            .map(Element::getValue)
+            .collect(Collectors.toMap(
+                PartyDetails::getSolicitorEmail,
+                i -> i.getRepresentativeFirstName() + " " + i.getRepresentativeLastName(),
```

Comment:
use common method

Reply:
used common method instead of individual method to get values